### PR TITLE
Daemon manager

### PR DESCRIPTION
**Pull request by Jaqueeee**
Start/stop daemon from GUI + random improvements.
Requires a monerod/monerod.exe in same folder as monero-core binary.
All cli arguments are passed to monerod. For example --data-dir

**Win32** 
Includes monerod 0.10.0.0
https://www.dropbox.com/s/ooigmm2kjxaja1t/monero-core-win32-daemon-manager-jaquee.zip?dl=1
**Note:** Sending funds is not is possible with included daemon. 

**OS X**
Includes monerod built from current master `17b6bd6`
https://www.dropbox.com/s/lqs9mpc50k20oq5/monero-core-osx-daemon-manager-jaquee.zip?dl=1

Fixes
https://github.com/monero-project/monero-core/issues/39
https://github.com/monero-project/monero-core/issues/165

### CODE REVIEW

File: src/daemon/DaemonManager.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#include <QUrl>
+#include <QProcess>
+
+class DaemonManager : public QObject
+{
+    Q_OBJECT
+
+public:
+
+    static DaemonManager * instance(QStringList args);
```
Comment (mbg033) [inline src/daemon/DaemonManager.h]: please use const reference for input arguments like this

File: src/libwalletqt/Wallet.cpp [hunk 2]
```diff
@@ -87,9 +88,20 @@ Wallet::Status Wallet::status() const
     return static_cast<Status>(m_walletImpl->status());
 }
 
-Wallet::ConnectionStatus Wallet::connected() const
+Wallet::ConnectionStatus Wallet::connected()
 {
-    return static_cast<ConnectionStatus>(m_walletImpl->connected());
+    // cache connection status
+    if(!m_initialized || m_connectionStatusTime.elapsed() / 1000 > m_connectionStatusTtl){
```
Comment (mbg033) [inline src/libwalletqt/Wallet.cpp]: Please don't hesitate to use blanks. https://wiki.qt.io/Qt_Coding_Style#Whitespace

File: src/libwalletqt/Wallet.cpp [hunk 3]
```diff
@@ -87,9 +88,20 @@ Wallet::Status Wallet::status() const
     return static_cast<Status>(m_walletImpl->status());
 }
 
-Wallet::ConnectionStatus Wallet::connected() const
+Wallet::ConnectionStatus Wallet::connected()
```
Comment (mbg033) [inline src/libwalletqt/Wallet.cpp]: It seems weird. Method supposed to be accessor but it actually changes state of the object.

File: src/daemon/DaemonManager.h (hunk 1 above)
Comment (Jaqueeee) [inline src/daemon/DaemonManager.h]: fixed

File: src/libwalletqt/Wallet.cpp (hunk 3 above)
Comment (Jaqueeee) [inline src/libwalletqt/Wallet.cpp]: reverted to const function and mutable variables.

File: src/libwalletqt/Wallet.cpp (hunk 2 above)
Comment (Jaqueeee) [inline src/libwalletqt/Wallet.cpp]: ok. thanks. will read up on that.

**Comment by dternyak** [comment]
The daemon in the dropbox binary isn't compatible with the GUI - you might have already known this but I wanted to mention it anyway.

**Comment by Jaqueeee** [comment]
@dternyak overlap issue fixed in https://github.com/monero-project/monero-core/pull/196

Official 0.10.0.0 monerod is included in the builds. Unfortunately not fully compatible. I.e you can't send funds.

**Comment by dternyak** [comment]
@Jaqueeee Looks like you have it covered on both counts!

**Comment by Jaqueeee** [comment]
@dternyak updated osx zip to include compatible daemon.

**Pull request merged**